### PR TITLE
Adding L1 and L2 regularization to models

### PR DESCRIPTION
**Pull request by TianyuDu**

### CODE REVIEW

File: torch_choice/utils/run_helper.py [hunk 1]
```diff
@@ -33,14 +33,18 @@ def run(model, dataset, batch_size=-1, learning_rate=0.01, num_epochs=5000):
...
             item_index = batch['item'].item_index if isinstance(model, NestedLogitModel) else batch.item_index
-            loss = model.negative_log_likelihood(batch, item_index)
+            # the model.loss returns negative log-likelihood + regularization term.
+            loss = model.loss(batch, item_index)
 
-            ll -= loss.detach().item()# * len(batch)
-            count += len(batch)
+            if e % (num_epochs // 10) == 0:
+                # record log-likelihood.
+                ll -= model.negative_log_likelihood(batch, item_index).detach().item() # * len(batch)
```
Comment (kanodiaayush) [inline torch_choice/utils/run_helper.py:41]: use the loss defined above here?

**Review by kanodiaayush — CHANGES_REQUESTED** [review]

File: torch_choice/utils/run_helper.py (hunk 1 above)
Comment (kanodiaayush) [reply torch_choice/utils/run_helper.py:41]: resolved, i made an error

**Review by kanodiaayush — APPROVED** [review]
